py_gtfs_rt_ingestion/lib/postgres_utils.py

```diff
@@ -182,14 +187,24 @@ def _rds_writer_process(metadata_queue: Queue) -> None:
             "metadata_insert", filepath=metadata_path
         )
         process_logger.log_start()
-        try:
-            with engine.begin() as cursor:
-                cursor.execute(insert_statement)
-            good_insert += 1
-            process_logger.log_complete()
-        except Exception as exception:
-            bad_insert += 1
-            process_logger.log_failure(exception)
+        for retry_attempt in range(retry_count):
+            try:
+                process_logger.add_metadata(retry_attempts=retry_attempt)
+                with engine.begin() as cursor:
+                    cursor.execute(insert_statement)
+
+            except Exception as exception:
+                if retry_attempt == retry_count - 1:
+                    bad_insert += 1
+                    process_logger.log_failure(exception)
+
+            else:
+                good_insert += 1
+                process_logger.log_complete()
+                break
+
+            # wait for gremlins to disappear
+            time.sleep(15)
```

Comment:
we want this wait after the write is succesfull or we hit all three attempts?

seems like it might be helpful to have after the exception before trying to write again?